### PR TITLE
[otbn/sw] Add missing OTBN ECDSA sign R value to test verification

### PR DESCRIPTION
**Pull request by TRJensen**
The ECDSA signature verification step requires the R value as input.
In the OTBN test code, a signing is done followed by a verification.
Because the correct R value already exists in OTBN memory,
we get a successful signature verification. However when clearing
OTBN memory between the signing and verification step, the
verification step fails. Adding the R value to verification step
solves problem. Fix for #5626

Signed-off-by: Thomas Rosted Jensen <thomas@rosted.net>

### CODE REVIEW

File: sw/device/tests/otbn/otbn_ecdsa_p256_test.c [hunk 1]
```diff
@@ -241,19 +279,24 @@ static void test_ecdsa_p256_roundtrip(void) {
   check_data("signature_r", signature_r, kExpectedSignatureR, 32);
   check_data("signature_s", signature_s, kExpectedSignatureS, 32);
 
+  // Cleanup OTBN variables.
+  p256_ecdsa_clean(&otbn_ctx);
```
Comment (imphil) [inline sw/device/tests/otbn/otbn_ecdsa_p256_test.c:283]: To make the test slightly more robust, could you change this and the other places where you use `p256_ecdsa_clean()` to code which clears the data memory completely? `CHECK(otbn_zero_data_memory(&otbn_ctx) == kOtbnOk)` should do the trick.

**Review by imphil — COMMENTED** [review]
Thanks a lot, this is a great first pull request! 

I have left one comment about the code inline.

And could you run clang-format to format the code according to our style guide? This will then make the automated continuous integration checks in this pull request green.

If you have made the changes to the code, do a `git add sw/device/tests/otbn/otbn_ecdsa_p256_test.c`, followed by a `git commit --amend` to update your commit. You can then do a `git push -f` to update the  `trj_fix_missing_r_value_in_ecdsa` branch in your fork, and with it this pull request.

**Comment by imphil** [comment]
@felixmiller can you have a quick look from an API perspective?

**Review by felixmiller — APPROVED** [review]
LGTM and thanks @TRJensen

File: sw/device/tests/otbn/otbn_ecdsa_p256_test.c (hunk 1 above)
Comment (TRJensen) [reply sw/device/tests/otbn/otbn_ecdsa_p256_test.c]: Great :) I have used this function instead. I noticed that I also need to reload the OTBN app again for verification.
Please have a look :)

**Review by imphil — APPROVED** [review]

**Comment by imphil** [comment]
Thank you for the updates, I just merged your changes.

Congratulations @TRJensen to your first ever pull request, and thanks for making OpenTitan your project of choice for such a memorable event :fireworks: :tada:

**Comment by TRJensen** [comment]
Thank you @imphil 🎉🎊☺️👍

**Pull request merged**